frontend/main.js

```diff
@@ -272,7 +339,14 @@ async function startApp () {
   }
 
   // register service-worker
-  await sbp('service-workers/setup')
+  await Promise.race(
+    [sbp('service-workers/setup'),
+      new Promise((resolve, reject) => {
+        setTimeout(() => {
+          reject(new Error('Timed out setting up service worker'))
```

Comment:
This should probably be a user-facing message that's wrapped in an `L` function and displayed to the user in an `alert`